### PR TITLE
Zmq

### PR DESCRIPTION
**Pull request by tfarago**
Adds *grab_send* which instead of giving images to the user sends them over network. Implemented in the base class and overloaded in uca-net becuase there we want the camera server to push the data, not us.

- [ ] Bump version to 2.4 (and release afterwards)

### CODE REVIEW

**Comment by matze** [comment]
That feels … wrong? Why not just add a client/server combo like `uca-net` (uhm … `uca-zmq`?).

**Comment by tfarago** [comment]
But that's the thing, the client is not defined on the uca level. What I need is to push the data down the network regardless the camera type, be it pco, mock or pco via net. This way I can completely avoid dealing with data in concert, where I just set up my experiment and tell the camera to push the data and have e.g. a writer or reconstruction clients waiting on the other end on a completely different machine so that my DAQ is not slowed down because all data goes through slow concert=python.
This functionality should be perhaps outside of any camera class because it is not a feature of a camera. But to have another wrapper plugin `uca-zmq` which would inside have an actual camera also seems strange. On the other hand, where else to put it so that I can easily access it from concert?

**Comment by matze** [comment]
> This functionality should be perhaps outside of any camera class because it is not a feature of a camera.

Exactly that. `ucad` is a "camera" server that wraps real camera plugins and pushes the data into a TCP socket. Couldn't you just have another `ucad-zmq` server that does the same but push the data via ZMQ instead?

**Comment by tfarago** [comment]
Sounds good, and I don't even need a separate server because I don't care if the camera control is done over plain sockets or zmq (I do need the control _and_ data stream if I want to synchronize things in concert). I can add the `grab_send` (or whatever better named) function to the existing `uca-net` which will not receive the data but tell `ucad` to send it elsewhere. Of course then this functionality would be limited to `uca-net`, but so what. So, something like
```python
camera = Camera('net')
camera.start_recording()
for i in range(10):
    camera.grab_send()
motor.stop()
```

**Comment by tfarago** [comment]
Or even cleaner instead of `camera.grab_send()` directly call the server from Python, which should be possible if that's really just a TCP socket, something like `ucad_socket.send('grab_send')`. And that would even allow native Python `asyncio` support.

**Comment by tfarago** [comment]
Or really a completely new `ucad-zmq` and pure Python implementation of the client in cocnert, plus side: all asyncio!

File: src/meson.build [hunk 1]
```diff
@@ -23,7 +23,8 @@ conf.set('GLIB_VERSION_MIN_REQUIRED', 'GLIB_VERSION_2_38')
 conf.set('GLIB_VERSION_MAX_ALLOWED', 'GLIB_VERSION_2_38')
 
 if python_dep.found()
-  conf.set('WITH_PYTHON_MULTITHREADING', true)
+  message(get_option('WITH_PYTHON_MULTITHREADING'))
```
Comment (tfarago) [inline src/meson.build:26]: o oh.

**Comment by tfarago** [comment]
Closing in favor of [uca-net#13](https://github.com/ufo-kit/uca-net/pull/13).

**Pull request closed without merging**